### ISSUE TITLE
Custom operation

### ISSUE THREAD

**Issue by suyizhang** [CarlosNZ/json-edit-react#110]
Hello boss, I'm here again. I tried the solution you provided earlier and it didn't quite meet the current needs. I pulled the source code to see if it's possible to add a parameter to this component to add custom action buttons  

Can you add parameters to increase the operation buttons within the EditButtons component

for example
![image](https://github.com/user-attachments/assets/a749c11c-788d-492b-a219-dd5dd58bf153)

**Comment by CarlosNZ** [comment]
Heya, I haven't forgotten this, just been busy.  I'm 90% sure you can make a custom operation button using a Custom Node component, but I wanted to check. Will get back to you shortly.

**Comment by CarlosNZ** [comment]
Okay, turns out that adding a custom wrapper is easy for the collection notes, but not currently that straightforward for the primitive value nodes.

So I've implemented the custom button as trial for you. It's available in https://www.npmjs.com/package/json-edit-react/v/1.15.10-rc1

I've left this completely undocumented for now, as it may change. See what you think.

### Usage

The prop is called `customButtons` and takes an array of the following:

```js
{
  Element: React.FC
  onClick: (nodeData: NodeData, e: React.MouseEvent) => void
}
```

So to add a new icon that prints the current value to the console you'd do something like:
```js
customButtons={[
    {
      Element: () => <MyIcon />,
      onClick: (nodeData) => console.log(nodeData.value),
    },
  ]}
}

Let me know how that works for you.

**Comment by suyizhang** [comment]
Very much the same, thank you for the help from the expert, and wish you all the best

**Comment by CarlosNZ** [comment]
Closing now, included in [1.15.10](https://www.npmjs.com/package/json-edit-react/v/1.15.10) release.
